File: day4/adventcoin.rb
require 'digest'

unless key = ARGV[0] #this is deliberate
  puts "please pass the puzzle input"
  exit
end

count = 1
while true
  coinhash = Digest::MD5.hexdigest(key + count.to_s)
  if coinhash[0,6] == '000000'
    puts count
    exit
  end
  count += 1
end

File: day4/adventcoin.py
from hashlib import md5

prefix = 'ckczppom'

num = 0
while True:
    num += 1
    input = prefix+str(num)
    hash = md5(bytes(input, 'utf-8'))
    if hash.hexdigest()[:5] == '00000':
        print(num)
        break

while True:
    num += 1
    input = prefix+str(num)
    hash = md5(bytes(input, 'utf-8'))
    if hash.hexdigest()[:6] == '000000':
        print(num)
        break